bitvm/src/signatures/public.rs

```diff
@@ -0,0 +1,838 @@
+        #[serde(
+            serialize_with = "serialize_byte_matrix_hex",
+            deserialize_with = "deserialize_byte_matrix_hex"
+        )]
+        compact_witness: Vec<Vec<u8>>,
+        #[serde(
+            serialize_with = "serialize_byte_matrix_hex",
+            deserialize_with = "deserialize_byte_matrix_hex"
+        )]
+        compact_signature: Vec<Vec<u8>>,
+    }
+
+    fn serialize_bytes_hex<S>(bytes: &[u8], serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&bytes.to_lower_hex_string())
+    }
+
+    fn deserialize_bytes_hex<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        Vec::<u8>::from_hex(&s).map_err(serde::de::Error::custom)
+    }
+
+    fn serialize_byte_matrix_hex<S>(
+        byte_matrix: &[Vec<u8>],
+        serializer: S,
+    ) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut seq = serializer.serialize_seq(Some(byte_matrix.len()))?;
+        for entry in byte_matrix {
+            seq.serialize_element(&entry.to_lower_hex_string())?;
+        }
+        seq.end()
+    }
+
+    fn deserialize_byte_matrix_hex<'de, D>(deserializer: D) -> Result<Vec<Vec<u8>>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let vec_s: Vec<String> = Vec::deserialize(deserializer)?;
+        vec_s
+            .into_iter()
+            .map(|s| Vec::<u8>::from_hex(&s).map_err(serde::de::Error::custom))
+            .collect()
+    }
+
+    #[allow(dead_code)]
+    fn create_test_vector<WOTS: Wots + CompactWots>(
+        message: &WOTS::Message,
+        secret_key: WinternitzSecret,
+    ) -> TestVector {
+        let public_key: Vec<Vec<u8>> = WOTS::generate_public_key(&secret_key)
+            .as_ref()
+            .iter()
+            .map(|entry| entry.to_vec())
+            .collect();
+        let witness: Vec<Vec<u8>> = WOTS::sign_to_raw_witness(&secret_key, message)
+            .iter()
+            .map(|entry| entry.to_vec())
+            .collect();
+        let signature: Vec<Vec<u8>> = WOTS::sign(&secret_key, message)
+            .as_ref()
+            .iter()
+            .map(|entry| entry.to_vec())
+            .collect();
+        let compact_witness: Vec<Vec<u8>> = WOTS::compact_sign_to_raw_witness(&secret_key, message)
+            .iter()
+            .map(|entry| entry.to_vec())
+            .collect();
+        let compact_signature: Vec<Vec<u8>> = WOTS::compact_sign(&secret_key, message)
+            .as_ref()
+            .iter()
+            .map(|entry| entry.to_vec())
+            .collect();
+
+        TestVector {
+            message: message.as_ref().to_vec(),
+            secret_key,
+            public_key,
+            witness,
+            signature,
+            compact_witness,
+            compact_signature,
+        }
+    }
+
+    #[allow(dead_code)]
```

Comment:
```suggestion
    #[expect(dead_code)]
```

If this thing is used, then this `expect` lint would fail.
I prefer that to `allow` which flies under the radar.